### DISCUSSION TITLE
How to read the absolute NTP timestamp from a Axis camera for each frame?

### DISCUSSION THREAD

**Discussion by christopher-wilke** [AxisCommunications/discussions#208] [VAPIX] [answered]
We need to get the absolute NTP timestamp for each frame from a AXIS camera (we are currently using RTSP). What's the best way to achieve this goal?

Right now, we are adding a text overlay to the stream. This text overlay gets then processed by a OCR engine.

Any feedback is highly appreciated.

**Comment by christopher-wilke** [comment]
Any idea?

**Reply by pppaulpeter** [reply, marked as answer]
combine rtp header and rtcp to caculate the absolute ntp timestamp

**Reply by christopher-wilke** [reply]
Can you recommend any library or share a sample? I researched a lot but havent found anything useful

**Reply by pppaulpeter** [reply]
> Can you recommend any library or share a sample? I researched a lot but havent found anything useful

you could ask chatgpt to solve the problem:)
